### ISSUE TITLE
⚠️Game crashes when booting [SOLVED]

### ISSUE THREAD

**Issue by User4716** [FlashyReese/sodium-extra-fabric#322]
### Make sure you are not opening a duplicate.

- [ ] [I have searched the issue tracker](https://github.com/FlashyReese/sodium-extra-fabric/issues?q=) and did not find an issue describing my crash.

### Minecraft version.

1.19.4

### Sodium Extra version.

sodium-extra-0.4.18+mc1.19.4-build.100.jar

### Sodium version.

sodium-fabric-mc1.19.4-0.4.11+build.26

### What happened?

Yesterday my game launched successfully, but today I updated Sodium Extra and now the game doesn't launch anymore. It just crashes and doesn't even reopen the launcher.

Below is the part of the Minecraft output log that caused my game to crash:

### Relevant logs

> [Render thread/ERROR]: Mixin apply for mod sodium-extra failed sodium-extra.mixins.json:sodium.biome_blend.MixinChunkRenderCache from mod sodium-extra -> me.jellysquid.mods.sodium.client.render.chunk.compile.pipeline.BlockRenderCache: org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException @Overwrite method createBiomeColorBlender in sodium-extra.mixins.json:sodium.biome_blend.MixinChunkRenderCache from mod sodium-extra was not located in the target class me.jellysquid.mods.sodium.client.render.chunk.compile.pipeline.BlockRenderCache. Using refmap sodium-extra-refmap.json
> org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: @Overwrite method createBiomeColorBlender in sodium-extra.mixins.json:sodium.biome_blend.MixinChunkRenderCache from mod sodium-extra was not located in the target class me.jellysquid.mods.sodium.client.render.chunk.compile.pipeline.BlockRenderCache. Using refmap sodium-extra-refmap.json
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachSpecialMethod(MixinPreProcessorStandard.java:436) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachOverwriteMethod(MixinPreProcessorStandard.java:416) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachMethods(MixinPreProcessorStandard.java:346) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attach(MixinPreProcessorStandard.java:299) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.createContextFor(MixinPreProcessorStandard.java:277) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinInfo.createContextFor(MixinInfo.java:1289) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:294) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.19.jar:?]
> at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
> at java.lang.Class.forName0(Native Method) ~[?:?]
> at java.lang.Class.forName(Class.java:467) ~[?:?]
> at net.fabricmc.loader.impl.launch.knot.MixinServiceKnot.findClass(MixinServiceKnot.java:96) ~[fabric-loader-0.14.19.jar:?]
> at org.spongepowered.asm.mixin.transformer.MixinProcessor.audit(MixinProcessor.java:254) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.audit(MixinTransformer.java:171) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.MixinEnvironment.audit(MixinEnvironment.java:1425) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at ca.fxco.memoryleakfix.fabric.MemoryLeakFixFabric.internalForceLoadAllMixinsAndClearSpongePoweredCache(MemoryLeakFixFabric.java:40) ~[memoryleakfix-fabric-1.17+-1.0.0.jar:?]
> at ca.fxco.memoryleakfix.fabric.MemoryLeakFixFabric.forceLoadAllMixinsAndClearSpongePoweredCache(MemoryLeakFixFabric.java:32) ~[memoryleakfix-fabric-1.17+-1.0.0.jar:?]
> at net.minecraft.class_310.handler$cnl000$memoryleakfix$memoryLeakFix$loadAllMixinsClientSide(class_310.java:17500) ~[client-intermediary.jar:?]
> at net.minecraft.class_310.method_1514(class_310.java) ~[client-intermediary.jar:?]
> at net.minecraft.client.main.Main.main(Main.java:237) ~[fabric-loader-0.14.19-1.19.4.jar:?]
> at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.19.jar:?]

‎ 
> [Render thread/ERROR]: Unhandled game exception
> java.lang.RuntimeException: Mixin transformation of me.jellysquid.mods.sodium.client.render.chunk.compile.pipeline.BlockRenderCache failed
> at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112) ~[fabric-loader-0.14.19.jar:?]
> at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
> at java.lang.Class.forName0(Native Method) ~[?:?]
> at java.lang.Class.forName(Class.java:467) ~[?:?]
> at net.fabricmc.loader.impl.launch.knot.MixinServiceKnot.findClass(MixinServiceKnot.java:96) ~[fabric-loader-0.14.19.jar:?]
> at org.spongepowered.asm.mixin.transformer.MixinProcessor.audit(MixinProcessor.java:254) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.audit(MixinTransformer.java:171) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.MixinEnvironment.audit(MixinEnvironment.java:1425) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
>at ca.fxco.memoryleakfix.fabric.MemoryLeakFixFabric.internalForceLoadAllMixinsAndClearSpongePoweredCache(MemoryLeakFixFabric.java:40) ~[memoryleakfix-fabric-1.17+-1.0.0.jar:?]
> at ca.fxco.memoryleakfix.fabric.MemoryLeakFixFabric.forceLoadAllMixinsAndClearSpongePoweredCache(MemoryLeakFixFabric.java:32) ~[memoryleakfix-fabric-1.17+-1.0.0.jar:?]
> at net.minecraft.class_310.handler$cnl000$memoryleakfix$memoryLeakFix$loadAllMixinsClientSide(class_310.java:17500) ~[client-intermediary.jar:?]
> at net.minecraft.class_310.method_1514(class_310.java) ~[client-intermediary.jar:?]
> at net.minecraft.client.main.Main.main(Main.java:237) ~[fabric-loader-0.14.19-1.19.4.jar:?]
> at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:462) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.19.jar:?]
> at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.19.jar:?]
> Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
> at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.19.jar:?]
> ... 18 more

‎ 
> Caused by: org.spongepowered.asm.mixin.throwables.MixinApplyError: Mixin [sodium-extra.mixins.json:sodium.biome_blend.MixinChunkRenderCache from mod sodium-extra] from phase [DEFAULT] in config [sodium-extra.mixins.json] FAILED during APPLY
> at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinError(MixinProcessor.java:638) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinProcessor.handleMixinApplyError(MixinProcessor.java:589) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:379) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.19.jar:?]
> ... 18 more

‎ 
> Caused by: org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: @Overwrite method createBiomeColorBlender in sodium-extra.mixins.json:sodium.biome_blend.MixinChunkRenderCache from mod sodium-extra was not located in the target class me.jellysquid.mods.sodium.client.render.chunk.compile.pipeline.BlockRenderCache. Using refmap sodium-extra-refmap.json
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachSpecialMethod(MixinPreProcessorStandard.java:436) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachOverwriteMethod(MixinPreProcessorStandard.java:416) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attachMethods(MixinPreProcessorStandard.java:346) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.attach(MixinPreProcessorStandard.java:299) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinPreProcessorStandard.createContextFor(MixinPreProcessorStandard.java:277) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinInfo.createContextFor(MixinInfo.java:1289) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinApplicatorStandard.apply(MixinApplicatorStandard.java:294) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.TargetClassContext.apply(TargetClassContext.java:421) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:403) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202) ~[sponge-mixin-0.12.4+mixin.0.8.5.jar:0.12.4+mixin.0.8.5]
> at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422) ~[fabric-loader-0.14.19.jar:?]
> ... 18 more

> [Render thread/INFO]: Stopping!

Then my game crashed

**Comment by User4716** [comment]
I thought that the new version might of broken something so I decided to downgrade the mod version.

But I found this upgraded version Build 101: https://github.com/FlashyReese/sodium-extra-fabric/releases/tag/mc1.19.3-0.4.19

And swapped the "broken" version to that and my game now works correctly.

**Comment by FlashyReese** [comment]
I marked that release as an alpha on CurseForge and Modrinth and archived it on CurseForge because Sodium 0.4.11 has been removed from Modrinth due to several issues with it.

**Comment by User4716** [comment]
It for some reason generates this mod version of sodium whenever I launch my game "sodium-fabric-mc1.16.5-0.2.0+build.4"
And I have 2 versions of sodium in my mods folder...

**Comment by User4716** [comment]
> I marked that release as an alpha on CurseForge and Modrinth and archived it on CurseForge because Sodium 0.4.11 has been removed from Modrinth due to several issues with it.

It showed as "Archived" on my CurseForge modpack, so I updated the mod and it wasn't showing as Archived anymore. It was now the Build 100.
Now I have the same error when starting up my game.

Sodium extra version was: sodium-extra-0.4.18+mc1.19.4-build.100
My fabric version is 0.14.19
Sodium version is sodium-fabric-mc1.19.4-0.4.11+build.26

And for some reason I also have this mod version of sodium in my mods folder, I deleted it yesterday but it came back
sodium-fabric-mc1.16.5-0.2.0+build.4

Edit: I deleted the 1.16.5 version and now only have the 1.19 version, but my game still crashes, I need to use the build 101 probably.
